### PR TITLE
refactor(!): gradually improve events

### PR DESCRIPTION
**Pull request by skaldarnar**
I recommend to go over this change set commit by commit.
	
This is technically a breaking change, but nothing in Omega seems to be using the affected API directly, so practically this change is local to the module. No module PRs are required.

### CODE REVIEW

**Comment by skaldarnar** [comment]
[Module Overhaul - Health](https://trello.com/c/zMUpksot/92-module-overhaul-health)

File: src/main/java/org/terasology/module/health/events/OnDamagedEvent.java [hunk 1]
```diff
@@ -11,20 +12,16 @@
...
 
-    public OnDamagedEvent() {
-    }
-
-    public OnDamagedEvent(int fullAmount, int change, Prefab damageType, EntityRef instigator) {
-        super(instigator, change);
-        this.fullAmount = fullAmount;
+    public OnDamagedEvent(int change, Prefab damageType, EntityRef instigator) {
+        super(-change, instigator);
+        Preconditions.checkArgument(change >= 0, "damage amount must be non-negative - use OnRestoredEvent instead");
```
Comment (jdrueckert) [inline src/main/java/org/terasology/module/health/events/OnDamagedEvent.java:19]: should this arg check be done before calling `super()`? :thinking:

File: src/main/java/org/terasology/module/health/events/OnRestoredEvent.java [hunk 2]
```diff
@@ -10,21 +11,14 @@
...
-    private int amount;
-
-    public OnRestoredEvent() {
-    }
 
     public OnRestoredEvent(int amount, EntityRef instigator) {
-        super(instigator, amount);
-        this.amount = amount;
+        super(amount, instigator);
+        Preconditions.checkArgument(amount >= 0, "restoration amount must be non-negative - use OnDamagedEvent instead");
```
Comment (jdrueckert) [inline src/main/java/org/terasology/module/health/events/OnRestoredEvent.java:17]: see above

File: src/main/java/org/terasology/module/health/events/RegisterRegenEvent.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.module.health.events;
+
+public class RegisterRegenEvent {
```
Comment (jdrueckert) [inline src/main/java/org/terasology/module/health/events/RegisterRegenEvent.java:6]: where is the regen registation processed? :thinking:

File: src/main/java/org/terasology/module/health/events/OnDamagedEvent.java (hunk 1 above)
Comment (skaldarnar) [reply src/main/java/org/terasology/module/health/events/OnDamagedEvent.java:19]: the call to `super()` needs to be the first statement in the constructor. I'd like to have the condition check first, but that's not possible.

File: src/main/java/org/terasology/module/health/events/RegisterRegenEvent.java (hunk 3 above)
Comment (skaldarnar) [reply src/main/java/org/terasology/module/health/events/RegisterRegenEvent.java:6]: nowhere yet. see commit message:

> This prepares to switch to a model and event flow better suited for the ECS and module collaboration for the regeneration feature. Both events are not used yet (neither sent nor reacted on).

File: src/main/java/org/terasology/module/health/events/OnDamagedEvent.java (hunk 1 above)
Comment (jdrueckert) [reply src/main/java/org/terasology/module/health/events/OnDamagedEvent.java:19]: okay, wasn't aware of that.

**Review by jdrueckert — APPROVED** [review]
Looks reasonable IMO. Feel free to wait for other reviewer's opinions.

**Pull request merged**